### ISSUE TITLE
Task Does Not Show

### ISSUE THREAD

**Issue by hediet** [microsoft/vscode#131626]
Testing #131402

```ts
import * as vscode from 'vscode';
import { ShellExecution, Task, TaskDefinition } from 'vscode';

class MyTaskDefiniton implements TaskDefinition {
    public readonly type = 'mytaskdef';
    constructor(public readonly detail: string) {}
}

export function activate(context: vscode.ExtensionContext) {
    const kind1 = new MyTaskDefiniton('foo');
    const kind2 = new MyTaskDefiniton('bar');

    vscode.tasks.registerTaskProvider('mytaskdef', {
        provideTasks: async () => {
            const task1 = new Task(kind1, vscode.TaskScope.Workspace, 'My Task Def 1', 'bla', 
            new ShellExecution('ls .'));
            task1.group = vscode.TaskGroup.Build;

            const task2 = new Task(kind2, vscode.TaskScope.Workspace, 'My Task Def 2', 'bla', 
            new ShellExecution('ls ..'));
            task2.group = vscode.TaskGroup.Build;

            return [
                task1,
                task2
            ];
        },
        resolveTask: async (task) => {
            return task;
        }
    });
    
    
    const disposable = vscode.commands.registerCommand('extension.helloWorld', async () => {
		const tasks = (await vscode.tasks.fetchTasks())
			.filter(task => (task.group?.id === vscode.TaskGroup.Build.id) && task.group?.isDefault);
		const taskNames = tasks.map(task => task.name).join(', ');
		vscode.window.showInformationMessage('Hello World, you have the following default tasks: ' + taskNames);
	});

    vscode.window.showInformationMessage('hello world');
}
```

![recording](https://user-images.githubusercontent.com/2931520/130771159-9ba29be9-85a2-4178-ae36-a9e08344de6d.gif)

**Comment by alexr00** [comment]
What's happening the gif for **Run Task**:
tl;dr: It's ✅
The `bla` tasks are contributed by an extension, which means that by default they do not show in the task picker at the top level _unless they have been used recently_. This is a perf feature. Instead, you must either "Show All Tasks", as is done in the gif, or choose the `Bla` task folder. The `bla` task folder will only show if the task provider is properly registered in the extensions package.json.

What's happening in the gif for **Configure default build task**:
This behavior looks wrong. We should be showing all `bla` tasks. But, I think this is just occurring because you don't have the task definition registered in the package.json. @hediet can you confirm whether you have the `taskDefinition` registered in your package.json?

**Comment by hediet** [comment]
This is my package.json:
```json
{
	"name": "example",
	"enableProposedApi": true,
	"private": true,
	"displayName": "Example",
	"description": "Example.",
	"version": "0.1.0",
	"publisher": "hediet",
	"readme": "./README.md",
	"engines": {
		"vscode": "^1.46.0"
	},
	"activationEvents": [
		"*"
	],
	"main": "./out/index",
	"contributes": {
		"taskDefinitions": [
			{
				"type": "mytaskdef"
				
			}
		]
	},
	"scripts": {
		"dev": "tsc --watch"
	},
	"devDependencies": {
		"@types/node": "^13.13.5",
		"@types/vscode": "*"
	},
	"dependencies": {
		"typescript": "^4.4.0-dev.20210728"
	}
}
```

**Comment by alexr00** [comment]
Hmm. I still cannot reproduce the issue, though now I see this which you have mentioned before:
![image](https://user-images.githubusercontent.com/38270282/130809256-40fc0514-f648-4d22-954b-7ef754f56574.png)

This is happening because the `mytaskdef` definition doesn't have any properties registered in the package.json.
